.github/workflows/main.yml

```diff
@@ -2,39 +2,62 @@ name: CI Workflow
 
 on:
   pull_request:
-    branches : [ develop ]
-  
-  workflow_dispatch: 
-  
+    branches: [develop]
+
+  workflow_dispatch:
+
   push:
 
 jobs:
   python-tests:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.8
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements/dev.txt
-        pip install pytest-django
-    - name: Run migrations
-      run: python project/manage.py migrate
-    - name: Run tests
-      env:
-        CIVIWIKI_LOCAL_NAME : True
-        DJANGO_SETTINGS_MODULE: core.settings
-      run: |
-        cd project
-        pytest -p no:warning
-    - name: Lint
-      uses: py-actions/flake8@v1
-      with:
-        ignore: "E203,W503"
-        max-line-length: "88"
-        exclude: "**/migrations/*.py"
-
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
```

Comment:
In your[ pyproject.toml ](https://github.com/CiviWiki/OpenCiviWiki/pull/1330/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R9)file, the Python version is defined as `^3.9`

Let's decide which Python version we will use. :) Also I wouldn't specify the exact Python version. Our pipelines constantly get broken because of the precise Python versions. You can use "^3.10" or "^3.9".

Reply:
I'll try ^3.9.

I don't want developers to be pinned down too tightly in their Python version, since they might not be familiar with `pyenv`.

Reply:
Looks like ^3.9 worked